nucliadb_node/tests/test_search_fields_filtering.rs

```diff
@@ -0,0 +1,154 @@
+    let mut fixture = NodeFixture::new();
+    fixture.with_writer().await?.with_reader().await?;
+    let mut writer = fixture.writer_client();
+    let mut reader = fixture.reader_client();
+
+    let new_shard_response = writer.new_shard(Request::new(NewShardRequest::default())).await?;
+    let shard_id = &new_shard_response.get_ref().id;
+
+    // Add a resource with vectors
+    let resource = build_resource_with_field(shard_id.clone(), "f/field1".to_string());
+    let result = writer.set_resource(resource).await?;
+
+    assert_eq!(result.get_ref().status(), nodewriter::op_status::Status::Ok);
+
+    // Search filtering with an unexisting field, to check that no vector are returned
+    let magnitude = f32::sqrt((17.0_f32).powi(2) * VECTOR_DIMENSION as f32);
+    let query_vector = vec![17.0 / magnitude; VECTOR_DIMENSION];
+
+    let search_request = nodereader::SearchRequest {
+        shard: shard_id.clone(),
+        fields: ["f/foobar".to_string()].to_vec(),
+        vector: query_vector,
+        page_number: 0,
+        result_per_page: 1,
+        min_score_semantic: -1.0,
+        paragraph: false,
+        document: false,
+        ..Default::default()
+    };
+    let results = reader.search(Request::new(search_request)).await?.into_inner();
+
+    assert!(results.vector.is_some());
+    let vector_results = results.vector.unwrap();
+    assert_eq!(vector_results.documents.len(), 0);
+
+    // Search filtering with a real field, to check that the vector is returned
+    let query_vector2 = vec![17.0 / magnitude; VECTOR_DIMENSION];
+    let search_request = nodereader::SearchRequest {
+        shard: shard_id.clone(),
+        vector: query_vector2,
+        fields: ["f/field1".to_string()].to_vec(),
+        page_number: 0,
+        result_per_page: 1,
+        min_score_semantic: -1.0,
+        paragraph: false,
+        document: false,
+        ..Default::default()
+    };
+    let results = reader.search(Request::new(search_request)).await?.into_inner();
+
+    assert!(results.vector.is_some());
+    let vector_results = results.vector.unwrap();
+    assert_eq!(vector_results.documents.len(), 1);
+
+    Ok(())
+}
+
+fn build_resource_with_field(shard_id: String, field_id: String) -> noderesources::Resource {
+    let rid = Uuid::new_v4().to_string();
+
+    let now = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).unwrap();
+    let timestamp = Timestamp {
+        seconds: now.as_secs() as i64,
+        nanos: 0,
+    };
+
+    let mut texts = HashMap::new();
+    texts.insert(
+        field_id.clone(),
+        noderesources::TextInformation {
+            text: "Dummy text".to_string(),
+            labels: vec![],
+        },
+    );
+
+    let mut field_paragraphs = HashMap::new();
+    for i in 1..=20 {
+        let mut sentences = HashMap::new();
+        sentences.insert(
+            format!("{rid}/{field_id}/{i}/paragraph-{i}"),
```

Comment:
Aren't vector ids more like `{rid}/{field_id}/{index}/{vstart}-{vend}`? I know it doesn't matter for the test but would be more correct